Octokit.Tests.Integration/GithubClientExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Octokit.Tests.Integration
+{
+    public static class GithubClientExtensions
+    {
+        public async static Task<DisposableRepository> CreateDisposableRepository(this IGitHubClient client, NewRepository newRepository)
+        {
+            return await ((TestRepositoriesClient)client.Repository).CreateDisposableRepository(newRepository);
```

Comment:
Rather than introducing extra abstractions for GitHubClient and RepositoryClient, you could do this inside `CreateDisposableRepository`:

```
var repository = await client.Repository.Create(newRepository);
return DisposableRepository.InitFromRepository(repository);
```